src/main/cpp/src/hive_hash.cu

```diff
@@ -486,16 +529,70 @@ std::unique_ptr<cudf::column> hive_hash(cudf::table_view const& input,
 
   check_nested_depth(input);
 
+  // `flattened_column_views` only contains nested columns and columns that result from flattening
+  // nested columns
+  std::vector<cudf::column_view> flattened_column_views;
```

Comment:
Oh this is wrong. The purpose of flattening columns is to avoid storing nested columns! If we need placeholder for the nested columns, we need to store empty nested columns instead of the original nested ones.